Даны две независимые выборки. Не соблюдается условие нормальности

x1  380, 420, 290

y1  140, 360, 200, 900

Сделайте вывод по результатам, полученным с помощью функции

In [2]:
# критерий Манна-Уитни

import numpy as np
import scipy.stats as stats

group_1 = np.array([380, 420, 290])
group_2 = np.array([140, 360, 200, 900])

stats.mannwhitneyu(group_1, group_2)

# Вывод: Так как p-value > 0.05 (5%), то нулевая гипотеза (данные не различаются) - принимается

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [3]:
# критерий Фридмана

import numpy as np
import scipy.stats as stats

A = np.array([150, 160, 165, 145, 155])
A_10 = np.array([140, 155, 150,  130, 135])
A_30 = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(A, A_10, A_30)

# Вывод: Так как p-value < 0.05 (5%), то нулевая гипотеза (есть ли статистически значимые различия) - не принимается (различия есть)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Сравните 1 и 2е измерения, предполагая, что 3го измерения через 30 минут не было.

In [4]:
# критерий Уилкоксона

import numpy as np
import scipy.stats as stats

A = np.array([150, 160, 165, 145, 155])
A_10 = np.array([140, 155, 150,  130, 135])

stats.wilcoxon(A, A_10)

# Вывод: Так как p-value > 0.05 (5%), то нулевая гипотеза (есть ли статистически значимые различия) - принимается (различий нет)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Даны 3 группы учеников плавания.

В первой группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа: 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

Выбрать тест и проверить, есть ли различия между выборками

In [5]:
# критерий Крускала-Уоллиса
import numpy as np
import scipy.stats as stats

gr1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr1, gr2, gr3)

# Вывод: Так как p-value > 0.05 (5%), то нулевая гипотеза (есть ли различия между выборками) - принимается (различий нет)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [11]:
# Нулевая гипотеза: партия изготавливается со средним арифметическим 2,5 см. (Если t-статистики меньше или равна t-критического)
# Альтернативная гипотеза: партия не изготавливается со средним арифметическим 2,5 см. (Если t-статистики больше t-критического)

import numpy as np
arr = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = 10
mu = 2.5

# Среднее арифметическое в партии
x = np.mean(arr)

# Cтандартное отклонение
s = np.std(arr, ddof = 1)

# t-статистики
print(f't-статистики = {(x - mu) / (s / n ** 0.5)}')

t-статистики = 0.5630613661802959


In [13]:
from scipy import stats
a = (1 - 0.05)

# t-критическое
t_tabl = stats.t.ppf(a, n - 1)
print(f't-критическое = {t_tabl}')

t-критическое = 1.8331129326536333


Вывод: т.к. t-статистики < t-критическое (0.56 < 1.83), то нулевая гипотеза (утверждение, что партия изготавливается со средним арифметическим 2,5 см) - принимается